test/acceptance/framework/helpers/consul.go

```diff
@@ -0,0 +1,44 @@
+package helpers
+
+import (
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/hashicorp/terraform-aws-consul-ecs/test/acceptance/framework/logger"
+	"github.com/stretchr/testify/require"
+)
+
+// WaitForConsulServices waits for services to show in the consul service catalog.
+func WaitForConsulServices(t *testing.T, consulClient *api.Client, serviceNames ...string) {
```

Comment:
These are all "naive" refactorings, and are a little limited (can't customize timeouts). I'm open to rethinking the interfaces here... Maybe a struct + methods that can be overridden, like

```go
type TestHelpers struct {
    t *testing.T
    suite *config.TestConfig
    consulClient *api.Client
    timer *retry.Timer
}

func NewTestHelpers(...) {
    return &TestHelpers{..., timer: nil}
}

func (h *TestHelpers) WithTimer(timer *retry.Timer) *TestHelpers {
    return &TestHelpers{...., timer: timer}
}

func (h *TestHelpers) WaitForConsulServices(serviceName ...string) {
    timer := h.timer
    if timer == nil {
         // Use default timer
        timer = &retry.Timer{Timeout: 6 * time.Minute, Wait: 20 * time.Second}
    }
    ...
}
```